# Clayton Seabaugh: Machine Learning Final Project
**Author:** Clayton Seabaugh  
**Date:** 4-20-2025  
**Objective:** This project demonstrates my ability to apply regression modeling techniques to a real-world dataset
- Load and explore a dataset
- Choose and justify features for predictign a target variable
- Train a regression model and evaluate performance
- Compare multiple regression approaches

## Section 1: Imports and Inspect the Data

### Section 1.1: Import modules and inspect first 10 rows

In [2]:
# Imports needed
import seaborn as sns
import pandas as pd
import matplotlib.pyplot as plt
import os


In [4]:
# Create file path for data and print head
file_path = 'data/insurance.csv'
if os.path.exists(file_path):
    # Read and print df
    df = pd.read_csv(file_path)
    print(df.head(10))
else:
    print(f"File {file_path} does not exist.")

   age     sex     bmi  children smoker     region      charges
0   19  female  27.900         0    yes  southwest  16884.92400
1   18    male  33.770         1     no  southeast   1725.55230
2   28    male  33.000         3     no  southeast   4449.46200
3   33    male  22.705         0     no  northwest  21984.47061
4   32    male  28.880         0     no  northwest   3866.85520
5   31  female  25.740         0     no  southeast   3756.62160
6   46  female  33.440         1     no  southeast   8240.58960
7   37  female  27.740         3     no  northwest   7281.50560
8   37    male  29.830         2     no  northeast   6406.41070
9   60  female  25.840         0     no  northwest  28923.13692


### Section 1.2: Check for missing values and display summary statistics

In [5]:
# Check for missing values
missing_values = df.isnull().sum()
print(f"Missing Values in each Column:\n{missing_values}")

# Generate summary statistics
summary_stats = df.describe()
print (f"Summary Statistics:\n{summary_stats}")

Missing Values in each Column:
age         0
sex         0
bmi         0
children    0
smoker      0
region      0
charges     0
dtype: int64
Summary Statistics:
               age          bmi     children       charges
count  1338.000000  1338.000000  1338.000000   1338.000000
mean     39.207025    30.663397     1.094918  13270.422265
std      14.049960     6.098187     1.205493  12110.011237
min      18.000000    15.960000     0.000000   1121.873900
25%      27.000000    26.296250     0.000000   4740.287150
50%      39.000000    30.400000     1.000000   9382.033000
75%      51.000000    34.693750     2.000000  16639.912515
max      64.000000    53.130000     5.000000  63770.428010


### Reflection 1: 
- There are 7 columns: age, sex, bmi, children, smoker, region, charges
- 3 Columns are categorical (sex, smoker, and region) and 4 are numerical(age, bmi, children, charges)
- There is a wide range of charges from 1121.87 min and 63770 max
- There are no null values in any of the columns and no data issues